src/mono/System.Private.CoreLib/src/System/Reflection/Metadata/MetadataUpdater.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Reflection.Metadata
+{
+    public static class MetadataUpdater
+    {
+        /// <summary>
+        /// Updates the specified assembly using the provided metadata, IL and PDB deltas.
```

Comment:
I wonder if we can make this a `partial` class and have all the stuff that is "common" between mono and coreclr shared in `src\libraries\System.Private.CoreLib`? The "public API" could be shared, and then call into the other partial class for the implementation.

Right now the XML doc is duplicated between both.

Reply:
The implementations of all 3 APIs are different between coreclr and mono. I don't see anything that would be common exception the XML doc headers.

Reply:
> The implementations of all 3 APIs are different between coreclr and mono

Maybe forcing things to be more common would lead to better code. I see the argument checking is roughly the same, but different. For example, mono checks if you are trying to apply an update to CoreLib, and throws `InvalidOperationException(SR.InvalidOperation_AssemblyNotEditable)` (which InvalidOperationException is not doc'd BTW). What does coreclr do?

Mono uses the better pattern of `if (assembly is not RuntimeAssembly runtimeAssembly)`.

Reply:
Other than argument checking (which I can fix the coreclr version) there isn't anything in common.  Coreclr doesn't need to explicitly check for System.Private.CoreLib because the QCall does that (and it does return InvalidOperationException).  

It really doesn't seem worth to have common code here.

I'll change the coreclr code to use the same argument checking as mono and add InvalidOperationException and ArgumentException to the XML header.